livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py

```diff
@@ -149,12 +184,145 @@ def __init__(
+
+    def update_options(
+        self,
+        *,
+        disable_partial_transcripts: Optional[bool] = None,
+        word_boost: Optional[List[str]] = None,
+        end_utterance_silence_threshold: Optional[int] = None,
+        enable_extra_session_information: Optional[bool] = None,
+        buffer_size_seconds: Optional[float] = None,
+    ):
+        if disable_partial_transcripts is not None:
+            self._opts.disable_partial_transcripts = disable_partial_transcripts
+        if word_boost is not None:
+            self._opts.word_boost = word_boost
+        if end_utterance_silence_threshold is not None:
+            self._opts.end_utterance_silence_threshold = end_utterance_silence_threshold
+        if enable_extra_session_information is not None:
+            self._opts.enable_extra_session_information = (
+                enable_extra_session_information
+            )
+        if buffer_size_seconds is not None:
+            self._opts.buffer_size_seconds = buffer_size_seconds
+
+        self._reconnect_event.set()
 
     async def _run(self) -> None:
         """
         Run a single websocket connection to AssemblyAI and make sure to reconnect
         when something went wrong.
         """
+
+        closing_ws = False
+
+        async def send_task(ws: aiohttp.ClientWebSocketResponse):
+            try:
+                nonlocal closing_ws
+
+                if self._opts.end_utterance_silence_threshold:
+                    await ws.send_str(
+                        json.dumps(
+                            {
+                                "end_utterance_silence_threshold": self._opts.end_utterance_silence_threshold
+                            }
+                        )
+                    )
+
+                samples_per_buffer = self._opts.sample_rate // round(
+                    1 / self._opts.buffer_size_seconds
+                )
+                audio_bstream = utils.audio.AudioByteStream(
+                    sample_rate=self._opts.sample_rate,
+                    num_channels=1,
+                    samples_per_channel=samples_per_buffer,
+                )
+
+                # forward inputs to AssemblyAI
+                # if we receive a close message, signal it to AssemblyAI and break.
+                # the recv task will then make sure to process the remaining audio and stop
+                async for data in self._input_ch:
+                    if isinstance(data, self._FlushSentinel):
+                        frames = audio_bstream.flush()
+                    else:
+                        frames = audio_bstream.write(data.data.tobytes())
+
+                    for frame in frames:
+                        self._speech_duration += frame.duration
+                        await ws.send_bytes(frame.data.tobytes())
+
+                closing_ws = True
+                await ws.send_str(SpeechStream._CLOSE_MSG)
+            except asyncio.CancelledError:
```

Comment:
I'm not sure why we want to catch the CancelledError?

Reply:
if reconnect event is set then these task will be cancelled. to avoid errors in logs I've added that

Reply:
updated code with removing those extra catches for CancelledError